[Bug]: Attribution generation fails

### Describe the bug

Generationg attributions fails since this commit: https://github.com/MetaMask/metamask-extension/commit/df3931ed76c2b9c497c3c6683305eadee3f706da

It fails because we're now installing foundry in our `postinstall` script. Specifically the crash is caused by the use of `tsx`, but generally we don't want to install foundry at all in this context. We're doing this in `postinstall` for ease of development, but it has this unwanted side-effect of breaking the attribution generation script.


### Expected behavior

The script should not crash.

### Screenshots/Recordings

_No response_

### Steps to reproduce

Run `yarn attributions:generate`

### Error messages or log output

```shell
node:internal/modules/cjs/loader:1228
  throw err;
  ^

Error: Cannot find module '/home/dan/metamask/metamask-extension/node_modules/tsx/dist/cli.mjs'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1225:15)
    at Module._load (node:internal/modules/cjs/loader:1051:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.18.0
```

### Detection stage

On main branch

### Version

main

### Build type

None

### Browser

Other (please elaborate in the "Additional Context" section)

### Operating system

Other (please elaborate in the "Additional Context" section)

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

--- a/development/generate-attributions.sh
+++ b/development/generate-attributions.sh
@@ -23,6 +23,9 @@ main() {
   # `plugin-allow-scripts.cjs` defined in the .yarnrc file
   export PATH="${SCRIPT_DIRECTORY}/generate-attributions/node_modules/.bin:${PATH}"
 
+  # Unset the root postinstall script to prevent it from installing devDependencies
+  node ./unset-postinstall.js
+
   # Switching to the project directory explicitly, so that we can use paths
   # relative to the project root irrespective of where this script was run.
   cd "${PROJECT_DIRECTORY}"
@@ -47,7 +50,7 @@ main() {
   if [ -z "${CI:-}" ]; then
     # If not running in CI, restore the allow-scripts plugin and development dependencies.
     cd "${PROJECT_DIRECTORY}"
-    git checkout -- .yarnrc.yml .yarn
+    git checkout -- .yarnrc.yml .yarn package.json
     yarn
   fi
 }

--- a/development/generate-attributions/unset-postinstall.js
+++ b/development/generate-attributions/unset-postinstall.js
@@ -4,15 +4,22 @@ const { readFile, writeFile } = require('node:fs/promises');
 const rootDirectory = path.resolve(__dirname, '..', '..');
 const packageJsonPath = path.join(rootDirectory, 'package.json');
 
+/**
+ * Unset the root `postinstall` script.
+ *
+ * This is used when generating attributions, to prevent development dependencies from being used
+ * or installed during `postinstall`.
+ */
 async function main() {
-  const packageJsonContents = await readFile(packageJsonPath, { encoding: 'utf8' });
+  const packageJsonContents = await readFile(packageJsonPath, {
+    encoding: 'utf8',
+  });
   const packageJson = JSON.parse(packageJsonContents);
   delete packageJson.scripts.postinstall;
   await writeFile(packageJsonPath, JSON.stringify(packageJson, null, 2));
 }
 
-main()
-  .catch((error) => {
-    console.error(error);
-    process.exitCode = 1;
-  })
+main().catch((error) => {
+  console.error(error);
+  process.exitCode = 1;
+});

--- a/development/generate-attributions/unset-postinstall.js
+++ b/development/generate-attributions/unset-postinstall.js
@@ -1,0 +1,18 @@
+const path = require('node:path');
+const { readFile, writeFile } = require('node:fs/promises');
+
+const rootDirectory = path.resolve(__dirname, '..', '..');
+const packageJsonPath = path.join(rootDirectory, 'package.json');
+
+async function main() {
+  const packageJsonContents = await readFile(packageJsonPath, { encoding: 'utf8' });
+  const packageJson = JSON.parse(packageJsonContents);
+  delete packageJson.scripts.postinstall;
+  await writeFile(packageJsonPath, JSON.stringify(packageJson, null, 2));
+}
+
+main()
+  .catch((error) => {
+    console.error(error);
+    process.exitCode = 1;
+  })

